packages/eas-cli/src/utils/__tests__/profiles-test.ts

```diff
@@ -94,3 +114,89 @@ describe(getProfilesAsync, () => {
     ).rejects.toThrowError(/eas.json is not valid/);
   });
 });
+describe(maybePrintBuildProfileDeprecationWarningsAsync, () => {
+  afterEach(() => {
+    clearHasPrintedDeprecationWarnings();
+    newLineSpy.mockClear();
+    warnSpy.mockClear();
+  });
+  const easJsonAccessor = EasJsonAccessor.fromProjectPath('/fake');
+  const readRawEasJsonMock = jest.spyOn(easJsonAccessor, 'readRawJsonAsync');
+  readRawEasJsonMock.mockImplementation(async () => {
+    return {};
+  });
+
+  describe('no deprecation warnings', () => {
+    it('does not print any warnings', async () => {
+      getBuildProfileDeprecationWarningsAsync.mockImplementation(async () => []);
+      const buildProfile = {} as BuildProfile<Platform.ANDROID>;
+      await maybePrintBuildProfileDeprecationWarningsAsync(buildProfile, easJsonAccessor);
+      expect(newLineSpy).not.toHaveBeenCalled();
+      expect(warnSpy).not.toHaveBeenCalled();
+    });
+  });
+  describe('one deprecation warning', () => {
+    it('prints the warning', async () => {
+      getBuildProfileDeprecationWarningsAsync.mockImplementation(async () => [
+        {
+          message: [
+            'The "build.production.cache.customPaths" field in eas.json is deprecated and will be removed in the future. Please use "build.production.cache.paths" instead.',
+          ],
+          docsUrl: 'https://docs.expo.dev/build-reference/eas-json/#cache',
+        },
+      ]);
+      const buildProfile = {} as BuildProfile<Platform.ANDROID>;
+      await maybePrintBuildProfileDeprecationWarningsAsync(buildProfile, easJsonAccessor);
+      expect(newLineSpy).toHaveBeenCalledTimes(2);
+      expect(warnSpy).toHaveBeenCalledTimes(3);
+      const warnCalls = warnSpy.mock.calls;
+      expect(warnCalls[0][0]).toEqual('Detected deprecated fields in eas.json:');
+      expect(warnCalls[1][0]).toEqual(
+        '\tThe "build.production.cache.customPaths" field in eas.json is deprecated and will be removed in the future. Please use "build.production.cache.paths" instead.'
+      );
+      const underlinedText = 'https://docs.expo.dev/build-reference/eas-json/#cache';
+      const dimmedText = `Learn more: ${chalk.underline(underlinedText)}`;
+      expect(warnCalls[2][0]).toEqual(`\t${chalk.dim(dimmedText)}`);
```

Comment:
Maybe it's worth considering using `.toMatchSnapshot` (https://jestjs.io/docs/snapshot-testing) instead of comparing every line of the output.

Reply:
But here I check with which arguments the `warn()` function has been called, how do I get the entire output to match against the snapshot?

Reply:
Bump